src/lib/consensus/proof_of_stake.ml

```diff
@@ -795,8 +867,19 @@ module Epoch_data = struct
         , prev_protocol_state_hash )
       else (curr_data.seed, curr_data.lock_checkpoint)
     in
+    let%map delegator = delegator_account curr_data in
+    let new_participated_stake =
+      if delegator.participated then curr_data.participated_stake
+      else
+        Option.value_exn
+          (Amount.add curr_data.participated_stake
+             (Balance.to_amount delegator.balance))
+    in
     let curr_data =
```

Comment:
The participated stake is being tracked in relation to the last epoch data, not the current. When this is lifted up directly to the consensus state, this should be fine though.